pkg/packetfilter/fake/packetfilter.go

```diff
@@ -67,117 +58,90 @@ func New() *PacketFilter {
 }
 
 func (i *PacketFilter) ChainExists(table packetfilter.TableType, chain string) (bool, error) {
-	return i.chainExists(table, chain)
+	return i.chainExists(uint32(table), chain)
 }
 
 func (i *PacketFilter) AppendUnique(table packetfilter.TableType, chain string, rule *packetfilter.Rule) error {
-	ruleSpecStr, err := json.Marshal(*rule)
-	if err != nil {
-		return errors.Wrap(err, "AppendUnique failed")
-	}
-
-	return i.addRule(table, chain, string(ruleSpecStr))
+	return i.addRule(table, chain, toRuleString(rule))
 }
 
 func (i *PacketFilter) CreateIPHookChainIfNotExists(chain *packetfilter.ChainIPHook) error {
-	if err := i.createChainIfNotExists(iphookChainTypeToTableType[chain.Type], chain.Name); err != nil {
-		return errors.Wrapf(err, "error creating IP tables %v:%s chain", iphookChainTypeToTableType[chain.Type], chain.Name)
-	}
-
-	return nil
+	return i.createChainIfNotExists(uint32(chain.Type), chain.Name)
 }
 
 func (i *PacketFilter) CreateChainIfNotExists(table packetfilter.TableType, chain *packetfilter.Chain) error {
-	if err := i.createChainIfNotExists(table, chain.Name); err != nil {
-		return errors.Wrapf(err, "error creating IP tables %s chain", chain.Name)
-	}
-
-	return nil
+	return i.createChainIfNotExists(uint32(table), chain.Name)
 }
 
 func (i *PacketFilter) DeleteIPHookChain(chain *packetfilter.ChainIPHook) error {
-	tableType := iphookChainTypeToTableType[chain.Type]
-	jumpRule := chain.JumpRule
-
-	if jumpRule == nil {
-		jumpRule = &packetfilter.Rule{
-			TargetChain: chain.Name,
-			Action:      packetfilter.RuleActionJump,
-		}
-	}
-
-	if err := i.Delete(tableType, chainHookToStr[chain.Hook], jumpRule); err != nil {
-		return errors.Wrap(err, "error deleting Jump Rule")
-	}
-
-	i.deleteChain(tableType, chain.Name)
+	i.deleteChain(uint32(chain.Type), chain.Name)
 
 	return nil
 }
 
 func (i *PacketFilter) DeleteChain(table packetfilter.TableType, chain string) error {
-	i.deleteChain(table, chain)
+	i.deleteChain(uint32(table), chain)
 
 	return nil
 }
 
 func (i *PacketFilter) ClearChain(table packetfilter.TableType, chain string) error {
-	i.clearChain(table, chain)
+	i.mutex.Lock()
```

Comment:
Overall code LGTM, 
I have one comment, can't we call deleteChain here?

Reply:
yeah so `DeleteChain` and `ClearChain` are the same in this implementation. I was wondering about that too. It seems `ClearChain` should just remove all the rules associated to the chain? If so then I can change that in a separate PR since I didn't change this functionality with this PR.